akka-stream-tests/src/test/scala/akka/stream/io/TlsSpec.scala

```diff
@@ -378,6 +378,36 @@ class TlsSpec extends StreamSpec("akka.loglevel=INFO\nakka.actor.debug.receive=o
       }
     }
 
+    "complete if TLS connection is truncated" in assertAllStagesStopped {
+
+      val ks = KillSwitches.shared("ks")
+
+      val scenario = SingleBytes
+
+      val outFlow: Flow[SslTlsOutbound, SslTlsInbound, NotUsed] = {
+        val terminator = BidiFlow.fromFlows(Flow[ByteString], ks.flow[ByteString])
+        clientTls(scenario.leftClosing) atop terminator atop serverTls(scenario.rightClosing).reversed join debug.via(scenario.flow) via debug
+      }
+
+      val inFlow = Flow[SslTlsInbound]
+        .collect { case SessionBytes(_, b) ⇒ b }
+        .scan(ByteString.empty)(_ ++ _)
+        .via(new Timeout(6.seconds))
+        .dropWhile(_.size < scenario.output.size)
+
+
+      val f =
+        Source(scenario.inputs)
+          .via(outFlow)
+          .via(inFlow)
+          .map(result => {
```

Comment:
Terminate connection after all data has been traveled through server flow and returned back to the client